### ISSUE TITLE
Consider removing `exitCode` from errors.

### ISSUE THREAD

**Issue by wtgtybhertgeghgtwtg** [lerna/lerna#1228]
The `exitCode` property on errors, as far as `lerna` goes, is always 1.  There's [additional work](https://github.com/lerna/lerna/blob/ee8dae1991f661007b0cca5994eb5c812ec93561/src/Command.js#L109-L119) to append it to errors.  If this is removed, `lerna` can use promises more internally instead of confusing and [deeply nested callbacks](https://github.com/lerna/lerna/blob/ee8dae1991f661007b0cca5994eb5c812ec93561/src/Command.js#L351-L363).

**Comment by evocateur** [comment]
The deeply nested callbacks are definitely on the chopping block. I rather like the idea of just calling init/exec from [p-try](https://github.com/sindresorhus/p-try) and having a single catch block.

There are still a few cases where we need to exit early _without_ a non-zero exit code ([`lerna publish`](https://github.com/lerna/lerna/blob/cdacb1a7a8d85d7e66eaa74dd9175b3cc03d04cb/src/commands/PublishCommand.js#L241) with nothing to do), as well as potentially exit early without error logging but _with_ a non-zero exit code ([`lerna updated`](https://github.com/lerna/lerna/blob/cdacb1a7a8d85d7e66eaa74dd9175b3cc03d04cb/src/commands/UpdatedCommand.js#L41) when used as a piped filter).

TL;DR: the `exitCode` was me making tests a bit easier (instead of checking/mutating `process.exitCode` all the time) for cases when "is an error" does not always correlate to "exited early". Yay complex legacy...

**Comment by wtgtybhertgeghgtwtg** [comment]
I can see that for `initialize` and `execute`, but it's a bit silly to check for an error code on an error when that code will always be 1.
